var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -72,10 +80,19 @@ def install(self, spec, prefix):
                     '-DLIBELF_INCLUDE_DIR=%s'   % join_path(
                         libelf.include, 'libelf'),
                     '-DLIBELF_LIBRARIES=%s'     % join_path(
-                        libelf.lib, 'libelf.so'),
-                    '-DLIBDWARF_INCLUDE_DIR=%s' % libdwarf.include,
-                    '-DLIBDWARF_LIBRARIES=%s'   % join_path(
-                        libdwarf.lib, 'libdwarf.so')]
+                        libelf.lib, 'libelf.so')]
+            if spec.satisfies('@:9.9.99'):
+                args.append('-DLIBDWARF_INCLUDE_DIR=%s' % libdwarf.include)
+                args.append('-DLIBDWARF_LIBRARIES=%s'   % join_path(
+                    libdwarf.lib, 'libdwarf.so'))
+            # For 10.0 + use elfutils libdw, libelf is an abstraction
+            # we are really using elfutils here
+            if spec.satisfies('@10.0.0:'):
+                args.append('-DLIBDWARF_INCLUDE_DIR=%s' % libelf.include)
+                args.append('-DLIBDWARF_LIBRARIES=%s'   % join_path(
+                    libelf.lib, 'libdw.so'))
```

Comment:
This won't work on macOS where the library name is `.dylib`, not `.so`. You should look at the `spec['libelf'].libs` attribute.

Reply:
@adamjstewart Thanks - I will check on how to change that.